src/options.ts

```diff
@@ -58,6 +58,7 @@ export interface Options {
   spanExporterFactory: SpanExporterFactory;
   spanProcessorFactory: SpanProcessorFactory;
   propagatorFactory: PropagatorFactory;
+  propagators: string;
```

Comment:
Should this perhaps be an array of `TextMapPropagator`? So when using the programmatic API (`startTracing({ ... })`) you'd be able to use a custom propagator.

Reply:
That's what `propagatorFactory` is for pretty much.
We actually went through a similar discussion with Owais on OTLP PR as well here: https://github.com/signalfx/splunk-otel-js/pull/218#discussion_r690267744
I'll give it another thought tomorrow and perhaps leave it as env-option only.

Reply:
Ok I see, only thing that might be somewhat quirky then is that if programmatic API is used and a custom propagator factory is set which returns a FooPropagator and `propagators` is set as well (e.g. to `tracecontext`), which combination of final propagators is expected? I.e. we then have 2 ways to configure the same thing, but they conflict somewhat.

Anyway this seems to be turning into bikeshedding, guess we can live with it, purely a hypothetical case :smile:

Reply:
Agree. I think we should stick with the factory only like we did in the other PR.

Reply:
> purely a hypothetical case 

I disagree Siim. These things confuse users and end up causing a lot of frustation and SWAT. No one wants more SWAT :)

Reply:
Good point, perhaps leaving it env-only and the default factory just checks the env var?

Reply:
Agree with both of you, **removed the option just like we did in OTLP PR**.
Sorry for that - I wrote this before I got any reviews on OTLP PR.